ke/tests/test_asm.py

```diff
@@ -125,3 +126,113 @@ def test_boundary():
     assert assemble(START_LINE) == r"^"
     assert assemble(START_STRING) == r"\A"
     assert assemble(WORD_BOUNDARY) == r"\b"
+
+
+def test_inline_flags():
+    save_PATTERN_IS_BYTES_LIKE = InlineFlag.PATTERN_IS_BYTES_LIKE
+    assert assemble(InlineFlag("a", None, Literal("test"))) == r"(?a:test)"
+    InlineFlag.PATTERN_IS_BYTES_LIKE = True
+    assert assemble(InlineFlag("L", None, Literal("test"))) == r"(?L:test)"
+    InlineFlag.PATTERN_IS_BYTES_LIKE = False
+    assert assemble(InlineFlag("u", None, Literal("test"))) == r"(?u:test)"
+    assert assemble(InlineFlag("i", None, Literal("test"))) == r"(?i:test)"
+    assert assemble(InlineFlag("m", None, Literal("test"))) == r"(?m:test)"
+    assert assemble(InlineFlag("s", None, Literal("test"))) == r"(?s:test)"
+    assert assemble(InlineFlag("i", "unset", Literal("test"))) == r"(?-i:test)"
+    assert assemble(InlineFlag("m", "unset", Literal("test"))) == r"(?-m:test)"
+    assert assemble(InlineFlag("s", "unset", Literal("test"))) == r"(?-s:test)"
+    assert (
+        assemble(InlineFlag("s", None, InlineFlag("s", None, Literal("test"))))
+        == r"(?s:test)"
+    )
+    assert (
+        assemble(InlineFlag("s", "unset", InlineFlag("s", "unset", Literal("test"))))
+        == r"(?-s:test)"
+    )
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("a", "unset", Literal("test")))
+    InlineFlag.PATTERN_IS_BYTES_LIKE = True
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("L", "unset", Literal("test")))
+    InlineFlag.PATTERN_IS_BYTES_LIKE = False
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("u", "unset", Literal("test")))
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("i", "garbage-nothing", Literal("test")))
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("i", "", Literal("test")))
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("a", None, InlineFlag("L", None, Literal("test"))))
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("L", None, InlineFlag("u", None, Literal("test"))))
+    with pytest.raises(CompileError):
+        assemble(InlineFlag("u", None, InlineFlag("a", None, Literal("test"))))
+    InlineFlag.PATTERN_IS_BYTES_LIKE = True
+    assert (
+        assemble(InlineFlag("L", None, InlineFlag("L", None, Literal("test"))))
+        == r"(?L:test)"
+    )
+
+    InlineFlag.PATTERN_IS_BYTES_LIKE = False
```

Comment:
pre-commit did not like my formatting of these tests, and made these really line-heavy trees. Originally each call was in it's own line.